### ISSUE TITLE
$file->move() always creates an empty 'index.html'

### ISSUE THREAD

**Issue by dusanbrankov** [codeigniter4/CodeIgniter4#8395]
### PHP Version

8.1

### CodeIgniter4 Version

4.0

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

MySQL

### What happened?

Whenever a file gets uploaded and saved with `$file->move()`, it doesn't only save the file in the specified directory, but it also creates an empty `index.html` file, every time.

I found [this issue](https://github.com/codeigniter4/CodeIgniter4/issues/4020), but appending a slash at the end of the path didn't solve my problem.

### Steps to Reproduce

Basically, all I am doing is saving the file with `move()`:

```php
$filename = $file->getRandomName();
$file->move("path/to/uploaddir", $filename);
```

### Expected Output

To just save the file without adding empty, unrelated files.

**Comment by michalsn** [comment]
This is by design - to increase security if you store files in a publicly accessible directory. It will prevent all files in the directory from being displayed (listed) if the server configuration allows it.

**Comment by dusanbrankov** [comment]
Oh, I see. I couldn't find that information in the documentation, so I thought it was a bug. Thanks for the clarification, I will close this issue then.
